File: src/server/AnkaCMS.Service/Models/PublicPartModel.cs
using System;
using System.Collections.Generic;
using AnkaCMS.Core.ValueObjects;

namespace AnkaCMS.Service.Models
{
    public class PublicPartModel
    {
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Keywords { get; set; }
        public DateTime LastModificationTime { get; set; }
        public Guid PartId { get; set; }
        public IdCodeName Language { get; set; }
        public List<PublicContentModel> Contents { get; set; }
    }
}

File: src/client/angular-web-public/src/app/models/public-part-model.ts
import { IdCodeName } from '../value-objects/id-code-name';
import { PublicContentModel } from './public-content-model';

export class PublicPartModel {
    code: string;
    name: string;
    description: string;
    keywords: string;
    lastModificationTime: Date;
    partId: string;
    language: IdCodeName;
    contents: PublicContentModel[];
    constructor() {
        this.language = new IdCodeName();
        this.contents = [];
    }
}
